spinta/manifests/xsd/helpers.py

```diff
@@ -0,0 +1,767 @@
+        # getting rid of the prefix
+        if ":" in property_type:
+            property_type = property_type.split(":")[1]
+
+        if property_type in self.xsd.custom_types:
+            property_type = self.xsd.custom_types.get(property_type).get("base", "")
+        if property_type in DATATYPES_MAPPING:
+            property_type = DATATYPES_MAPPING[property_type]
+        else:
+            property_type = "string"
+
+        return property_type
+
+    def _get_enums(self, node: _Element) -> dict:
+        enums = {}
+        simple_type = node.xpath(f'./*[local-name() = "simpleType"]')
+        if simple_type:
+            enums = self.xsd.get_enums_from_simple_type(simple_type[0])
+        else:
+            node_type = node.get("type")
+            if node_type and ":" in node_type:
+                node_type = node_type.split(":")[1]
+            if node_type in self.xsd.custom_types:
+                enums = self.xsd.custom_types[node_type]["enums"]
+
+        return enums
+
+    def _node_to_partial_property(self, node: etree.Element) -> tuple[str, dict]:
+        """Node can be either element or attribute.
+        This function only processes things common to attributes and elements"""
+        prop = dict()
+
+        prop["description"] = XSDReader.get_description(node)
+        property_name = node.get("name")
+        prop["external"] = {"name": property_name}
+        property_id = to_property_name(property_name)
+        prop["type"] = self._get_property_type(node)
+        if ";" in prop["type"]:
+            prop_type, target, value = prop["type"].split(";")
+            prop["type"] = prop_type
+            if target == "enum":
+                prop[target] = value
+            if target == "prepare":
+                prop["external"][target] = Expr(value)
+
+        prop["enums"] = self._get_enums(node)
+
+        return self.deduplicate(property_id), prop
+
+    def attributes_to_properties(self, element: etree.Element) -> dict:
+        properties = {}
+        attributes = element.xpath(f'./*[local-name() = "attribute"]')
+        complex_type = element.xpath(f'./*[local-name() = "complexType"]')
+        if complex_type:
+            properties.update(self.attributes_to_properties(complex_type[0]))
+        for attribute in attributes:
+
+            property_id, prop = self._node_to_partial_property(attribute)
+            if not prop["type"]:
+                prop["type"] = "string"
+            # property source
+            prop["external"]["name"] = f'@{prop["external"]["name"]}'
+
+            # property required or not. For attributes only.
+            use = attribute.get("use")
+            if use == "required":
+                required = True
+            else:
+                required = False
+            prop["required"] = required
+            properties[property_id] = prop
+
+        # todo attribute can be a ref to an externally defined attribute also. Not in RC though
```

Comment:
```suggestion
        # TODO: attribute can be a ref to an externally defined attribute also. Not in RC though
```

Žiūrėti: https://peps.python.org/pep-0350/